### PR TITLE
Update documentation for how to cut a release AND perform the action …

### LINKED ISSUE #37: Publish a first release
In the past, semantic conventions were released with the specifications, on a regular basis (monthly).

Language repositories, like opentelemetry-cpp, [followed][semconv-cpp] this schedule to release up to date C++ semantic conventions, generated from the specification yaml files.

Now that the semantic conventions moved to this repository, when code gets stable, please make a first release with a tag, so that:
* language repositories can adjust code generations scripts
* language repositories can stay up to date

Please also indicate the expected release frequency for semantic conventions (still monthly ?), so downstream repositories can plan accordingly.

[semconv-cpp]: https://github.com/open-telemetry/opentelemetry-cpp/issues?q=is%3Aissue+semantic+conventions+is%3Aclosed+

### PR DESCRIPTION
**Pull request by jsuereth**
…for the 1.21.0 release.   This will become the commit we tag for 1.21.

Fixes #37 #136 

Note: I think there's some better automation and checks we can do for `schema-next.yaml` and setting the version numbers on release going forward.

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM, https://github.com/open-telemetry/semantic-conventions/pull/190/files#r1261832861 doesn't have to be a blocker for the release cut.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ release.
 
 ## Unreleased
 
+## v1.21.0 (2023-07-12)
```
[the remark this answers is not in the document]
Comment (arminru) [reply CHANGELOG.md:10]:
```suggestion
## v1.21.0 (2023-07-13)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -122,3 +122,18 @@ make misspell-correction
...
+   [tooling to update the spec](#updating-the-referenced-specification-version)
+   if needed.
+2. Create a staging branch for the release.
+  a. Update `schema-next.yaml` file and move to `schemas/{version}`
+    - Ensure the `next` version is appropriately configured as the `{version}`.
+    - Copy `schema-next.yaml` to `schemas/{version}`.
+    - Add `next` as a version in `schema-next.yaml` version.
+  b. Update `CHANGELOG.md` for the latest version.
+    - Add `## v{version} ({date})` under `## Unreleased`
+  c. Send staging tag as PR for review.
```
Comment (arminru) [inline CONTRIBUTING.md:138]:
```suggestion
  - a. Update `schema-next.yaml` file and move to `schemas/{version}`
    - Ensure the `next` version is appropriately configured as the `{version}`.
    - Copy `schema-next.yaml` to `schemas/{version}`.
    - Add `next` as a version in `schema-next.yaml` version.
  - b. Update `CHANGELOG.md` for the latest version.
    - Add `## v{version} ({date})` under `## Unreleased`
  - c. Send staging tag as PR for review.
```

Otherwise it looks like this:
image

File: internal/tools/schema_check.sh [hunk 3]
```diff
@@ -32,12 +32,15 @@ grep -o -e '## v[1-9].*\s' $root_dir/CHANGELOG.md | grep -o '[1-9].*' | while re
...
 
-  curl --no-progress-meter https://opentelemetry.io/schemas/$ver > verify$ver
-
-  diff verify$ver $file && echo "Published schema at https://opentelemetry.io/schemas/$ver is correct" \
-    || (echo "Published schema at https://opentelemetry.io/schemas/$ver is incorrect!" && exit 3)
-
-  rm verify$ver
+  # Schema file will no be served directly from this repository when linked
+  # into opentelemetry.io.  We disable this for now and need to move the check
+  # into the website.
```
[the remark this answers is not in the document]
Comment (jsuereth) [reply internal/tools/schema_check.sh:37]: Because our changelog NOW includes a version not yet served. It won't be served until I tag this repository.

I opened a bug to shuffle around this logic.  Specifically, this PR would fail with `1.21.0` not being available on the website if I enabled this.

[the remark this answers is not in the document]
Comment (jsuereth) [reply internal/tools/schema_check.sh:37]: See #191

[the remark this answers is not in the document]
Comment (jsuereth) [reply internal/tools/schema_check.sh:37]: Going to resolve this.

TL;DR: is that previously I think we wouldn't bump the CHANGELOG.md as part of the release, leading to a release tag where CHANGELOG had an `## unreleased` header.   The release workflow I'm using now involves the tag having up-to-date SCHEMA file + `CHANGELOG`, however we are using the presence of a version in CHANGELOG to check the opentelemetry.io hosting.   There will now be a delay between cutting a release, bumping the CHANGELOG and opentelemetry.io hosting that previously we didn't have.

We'll figure out a new mechanism for testing this out as part of #191
